src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,130 @@
+package duke.tasks;
+
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import duke.exceptions.DukeSaveLoadException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A task.
+ */
+public abstract class Task {
+    /** The description of the task. */
+    public final String description;
+    /** Whether the task is done. */
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks the task as not done.
+     */
+    public void markAsNotDone() {
+        this.isDone = false;
+    }
+
+    /**
+     * Encodes an array of arbitrary values into a single-line string for 
+     * saving into a file.
+     * Since the delimiter " | " is used in the encoding, any vertical bar "|" 
+     * in the values is escaped.
+     * 
+     * @param values The array of values to encode.
+     * @return The encoded values.
+     */
+    protected static String encodeValues(String[] values) {
```

Comment:
I really like this idea. I didn't think about the implications of using | in encoding.